src/logsearch/logsearch.py

```
@@ -249,11 +247,18 @@ def search(self, query='', log_ids=None, time_range=None, from_time=None, to_tim
         :type time_range: basestring
         :type from_time: int
         :type to_time: int
-        :type progress: bool
+        :type show_progress: bool
         :type limit: int
         :type query_params: dict
         :rtype: LogSearch.Query
         """
+        if show_progress:
+            try:
+                import tabulate
+                from progress.bar import Bar
+            except ImportError as e:
+                show_progress = False # Progress cannot be printed, so the flag is unset
+                print 'Visual progress indication is not available.', e
```

We should throw an exception here I think

---

tbd